drivers/regulator/regulator_rpi_pico.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2023 TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_core_supply_regulator
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/dt-bindings/regulator/rpi_pico.h>
+#include <zephyr/sys/linear_range.h>
+#include <zephyr/toolchain.h>
+#include <hardware/regs/vreg_and_chip_reset.h>
+#include <hardware/structs/vreg_and_chip_reset.h>
+
+static const struct linear_range core_ranges[] = {
+	LINEAR_RANGE_INIT(800000u, 0u, 0u, 5u),
+	LINEAR_RANGE_INIT(850000u, 50000u, 6u, 15u),
+};
+
+struct regulator_rpi_pico_config {
+	struct regulator_common_config common;
+	const struct linear_range *ranges;
+	const size_t num_ranges;
```

Comment:
note: it looks like you only have one regulator (CORE), so there's no need to store ranges in the instance config struct. Just use `core_ranges` directly.

Reply:
Make it use `core_ranges` and `num_core_ranges` directly.